algoliasearch-common/src/main/java/com/algolia/search/objects/QueryBase.java

```diff
@@ -298,6 +303,63 @@
     return builder.put(name, String.join(",", attributes));
   }
 
+  private ImmutableMap.Builder<String, String> addListString(
+      ImmutableMap.Builder<String, String> builder, String name, List<List<String>> attributes) {
+
+    if (attributes == null) {
+      return builder;
+    }
+
+    String values =
+        "["
+            + attributes
+                .stream()
+                .map(
+                    arr ->
+                        "["
+                            + arr.stream()
+                                .map(QueryBase::formatParameters)
+                                .collect(Collectors.joining(","))
+                            + "]")
+                .collect(Collectors.joining(","))
+            + "]";
```

Comment:
I don't know if this style is enforced by `coveo` but it is particularly difficult to read. May we try to change that?

Reply:
It's coveo styling convention and I couldn't find a way to change it.